src/panel/direction/DirectionPanel.jsx

```diff
@@ -213,25 +213,29 @@ export default class DirectionPanel extends React.Component {
       previewRoute={activePreviewRoute}
     />;
 
-    if (Device.isMobile()) {
-      return <div className="direction_panel_mobile">
-        <div className="itinerary_close_mobile" onClick={this.onClose}>
-          <span className="icon-chevron-left" />
-          {_('return', 'direction')}
-        </div>
-        {title}
-        {!activePreviewRoute && form}
-        {result}
-      </div>;
-    }
-
-    return <Panel
-      title={title}
-      close={this.onClose}
-      className="direction_panel"
-    >
-      {form}
-      {result}
-    </Panel>;
+    return <DeviceContext.Consumer>
```

Comment:
Probably a matter of taste, but don't you think the new  [`useContext()`](https://fr.reactjs.org/docs/hooks-reference.html#usecontext) is more readable ?

Reply:
I agree, but following a discussion with @bbecquet , this would imply a migration of the complete class to a function component

Reply:
Ah ok, I thought it could be used directly in the `render()` function. Never mind.